caas/kubernetes/provider/bootstrap.go

```diff
@@ -544,10 +550,57 @@ func (c controllerStack) createControllerStatefulset() error {
 		logger.Debugf("deleting %q statefulset", spec.Name)
 		c.broker.deleteStatefulSet(spec.Name)
 	})
-	return errors.Trace(c.broker.createStatefulSet(spec))
+	w, err := c.broker.WatchUnits(c.resourceNameStatefulSet)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	defer w.Kill()
+
+	if err = c.broker.createStatefulSet(spec); err != nil {
+		return errors.Trace(err)
+	}
+	if err = c.syncPodStatus(w); err != nil {
+		return errors.Annotate(err, "fetching pods status for controller")
+	}
+	return nil
+}
+
+func (c *controllerStack) syncPodStatus(w watcher.NotifyWatcher) error {
+	printedMsg := set.NewStrings()
+	checkEvents := func(events []core.Event) bool {
+		if len(events) == 0 {
+			return false
+		}
+		startedContainers := 0
+		for _, evt := range events {
+			if evt.Type == core.EventTypeNormal && !printedMsg.Contains(evt.Message) {
+				printedMsg.Add(evt.Message)
+				logger.Debugf(evt.Message)
+			}
+			if evt.Reason == kubeletevents.StartedContainer {
+				logger.Debugf("event: %q, %q", evt.Reason, evt.Message)
+				startedContainers++
+			}
+		}
+		return startedContainers >= c.containerCount
+	}
+	for {
+		select {
+		case <-w.Changes():
+			events, err := c.broker.getEvents(getPodName(c.pcfg))
+			if err != nil {
+				return errors.Trace(err)
+			}
+			if checkEvents(events) {
+				// all containers are created.
+				c.ctx.Infof("Pulled images, created %d containers", c.containerCount)
```

Comment:
We don't need the "pulled images bit" as if the containers are created,the images must have been pulled.

Immediately after this, we print "Bootstrap agent now started". So "created 2 containers" doesn't really add anything to the user experience.
What we want is to inform the user what is happening between
Creating k8s resources for controller "controller-foo"
and
Bootstrap agent now started

Something like:

Creating k8s resources for controller "controller-foo"
Downloading pod images
Starting controller pod
Bootstrap agent now started

What we would do is print "Starting controller pod" at the start of the syncPodStatus() and block until things are started.
Ideally we'd also check for image downloading to be finished now that we can poll events etc - we log messages to that effect

Reply:
Now the logs are like 

```
Creating Juju controller "k2" on microk8s/localhost
Creating k8s resources for controller "controller-k2"
Starting controller pod
Bootstrap agent now started
Contacting Juju controller at 10.152.183.50 to verify accessibility...

Bootstrap complete, controller "k2" now is available in namespace "controller-k2"

Now you can run
	juju add-model <model-name>
to create a new model to deploy k8s workloads.
```